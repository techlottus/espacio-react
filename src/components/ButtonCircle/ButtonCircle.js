import React, { useEffect } from "react";

export const ButtonCircle = React.memo(({ data, onClick }) => {
  const buttoncircleRef = React.createRef();

  useEffect(() => {
    buttoncircleRef.current.data = {
      id: data.id || "",
      size: data.size || "",
      icon: data.icon || "",
      disabled: data.disabled,
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps


  return (
    <>
      <lottus-button-circle ref={buttoncircleRef}></lottus-button-circle>
    </>
  );
});
