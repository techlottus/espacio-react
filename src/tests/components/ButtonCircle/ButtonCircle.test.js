import React from "react";
import { mount } from "enzyme";
import { ButtonCircle } from "../../../components/ButtonCircle/ButtonCircle";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
  data: {
    id: null,
    type: null,
    title: null,
    size: null,
    icon: null,
    lyIcon: null,
    disabled: null,
    isExpand: null,
  },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <ButtonCircle data={initState.data} />
    </Provider>
  );
};

describe("Test <ButtonCircle />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
