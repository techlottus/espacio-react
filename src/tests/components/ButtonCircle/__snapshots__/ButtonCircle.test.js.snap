// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ButtonCircle /> should init component 1`] = `<lottus-button-circle />`;
